ui.apps/src/main/javascript/minicart/src/utils/state.js

```diff
@@ -0,0 +1,29 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+import React, { createContext, useContext, useReducer } from 'react';
+import { object, func } from 'prop-types';
+
+export const CartContext = createContext();
+
+export const CartProvider = ({ reducer, initialState, children }) => (
```

Comment:
This context provider doesn't contain any cart specific data, so I don't see a point in calling it `CartProvider`. However, I think we should also include the handler functions here so that any component wrapped in this provider can make use of them.

Reply:
I added the handler functions to `CartInitializer`. Since I added the reducers and default state to this file, I guess we can keep calling it `CartProvider`.